client/src/main/java/de/qabel/desktop/daemon/management/TransactionGroup.java

```diff
@@ -134,8 +137,59 @@ public long totalElements() {
     @Override
     public long finishedElements() {
         final long[] finished = {0};
-        visitAll(t -> { if (t.isDone()) { finished[0]++;}});
+        visitAll(t -> {
+            if (t.isDone()) {
+                finished[0]++;
+            }
+        });
 
         return finished[0];
     }
+
+    private Boolean isRemote(Transaction transaction) {
+        if (transaction instanceof Upload) {
+            return true;
+        }
+        return false;
+    }
+
+    private int countAllFiles(Boolean allFiles) {
+        synchronized (transactions) {
+            transactionsList = new ArrayList<>(transactions);
+            filesList = new LinkedList<>();
+            foundFile = false;
+            if (totalElements() > 0) {
+                filesList.add(transactionsList.get(0));
+                for (int i = 1; i < transactionsList.size(); i++) {
+                    for (int j = 0; j < filesList.size(); j++) {
+                        if (transactionsList.get(i).getDestination().equals(filesList.get(j).getDestination()) &&
```

Comment:
Iterate with a foreach instead.
Also, this probably is a major performance issue since the algorithm is O(n²). A faster algorithm would be for example:
```
M is a map
foreach transaction:
  if (destination, filename) in map: skip
  else:  M[(destination, filename)] = transaction
return M.size
```